cocotb/build_libs.py

```diff
@@ -0,0 +1,471 @@
+# Copyright cocotb contributors
+# Licensed under the Revised BSD License, see LICENSE for details.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import os
+import sys
+import sysconfig
+import errno
+import distutils
+import shutil
+import logging
+
+from setuptools import Extension
+from setuptools.dist import Distribution
+from distutils.spawn import find_executable
+from setuptools.command.build_ext import build_ext as _build_ext
+
+
+# Needed for Windows to not assume python module (generate interface in def file)
+class build_ext(_build_ext):
+    def get_export_symbols(self, ext):
+        return None
+
+
+def _rename_safe(target, link_name):
+    """Rename or symlink on Mac or copy on Windows."""
+
+    if sys.platform == "darwin":  # On Mac there is an issue with rename? Workaround!
+        try:
+            os.symlink(target, link_name)
+        except OSError as e:
+            if e.errno == errno.EEXIST:
+                os.remove(link_name)
+                os.symlink(target, link_name)
+            else:
+                raise e
+        return
+
+    if (
+        os.name == "nt"
+    ):  # On Windows there is an issue with symlink and rename? !Workaround!
+        shutil.copy2(target, link_name)
+        return
+
+    try:
+        os.rename(target, link_name)
+    except OSError as e:
+        if e.errno == errno.EEXIST:
+            os.remove(link_name)
+            os.rename(target, link_name)
+        else:
+            raise e
+
+
+def _build_lib(lib, dist, build_dir):
+    """Use setuptools to build *lib* into *build_dir*."""
+
+    dist.ext_modules = [lib]
+
+    _build_ext = build_ext(dist)
+    _build_ext.build_lib = build_dir
+    _build_ext.build_temp = build_dir
+
+    _build_ext.finalize_options()
+
+    _build_ext.run()
+    out_lib = _build_ext.get_outputs()
+
+    lib_name = lib.name
+    lib_path = os.path.abspath(out_lib[0])
+    dir_name = os.path.dirname(lib_path)
+    ext_name = os.path.splitext(lib_path)[1][1:]
+    if os.name == "nt":
+        ext_name = "dll"
+
+    target = os.path.join(os.path.abspath(dir_name), lib_name + "." + ext_name)
+    if target != lib_path:
+        _rename_safe(
+            lib_path, os.path.join(os.path.abspath(dir_name), lib_name + "." + ext_name)
+        )
+
+    return lib_name + "." + ext_name
+
+
+def _extra_link_args(lib_name):
+    if sys.platform == "darwin":
+        return ["-Wl,-install_name,@loader_path/%s.so" % lib_name]
+    else:
+        return []
+
+
+def build_common_libs(build_dir, include_dir, share_lib_dir, dist):
+
+    if sys.platform == "darwin":
+        ld_library = sysconfig.get_config_var("LIBRARY")
+    else:
+        ld_library = sysconfig.get_config_var("LDLIBRARY")
+
+    if ld_library:
```

Comment:
```suggestion
    if ld_library is not None:
```
(https://docs.python.org/3/library/sysconfig.html#sysconfig.get_config_var, returns ``None`` if not found)